File: test/table-base64/dart/main.dart
import 'dart:convert';
import 'dart:io';
import 'dart:typed_data';

import '../../../build/table-base64/dart/BytesTable.dart';
import '../../../build/table-base64/dart/Base64testTable.dart';

String hex(List<int> data) => data.isEmpty
    ? '-'
    : data.map((b) => b.toRadixString(16).padLeft(2, '0')).join();

Future<void> main() async {
  await for (final line
      in stdin.transform(utf8.decoder).transform(const LineSplitter())) {
    final text = Uint8List.fromList([
      for (var i = 0; i < line.length; i += 2)
        int.parse(line.substring(i, i + 2), radix: 16),
    ]);
    final value = Blob();
    final report = TableReport();
    value.fromJson(text, report);
    if (report.malformed) {
      print('1 0 0 - -');
      continue;
    }
    final output = Uint8List(value.toJsonMeasure());
    if (value.toJson(output) != output.length) {
      throw StateError('writer size');
    }
    print(
      '0 ${report.clamped} ${report.kindMismatch} ${hex(value.payload.sublist(0, value.payloadLength))} ${hex(output)}',
    );
  }
}

File: test/table-base64/elixir/main.exs
base = Path.expand("../../../build/table-base64/elixir", __DIR__)
Code.require_file("TableRuntime.ex", base)
Code.require_file("BytesTable.ex", base)

hex = fn
  <<>> -> "-"
  data -> Base.encode16(data, case: :lower)
end

for line <- IO.stream(:stdio, :line) do
  text = Base.decode16!(String.trim(line), case: :mixed)
  {value, report} = Base64test.BytesTable.from_json_blob(text)

  if report.malformed do
    IO.puts("1 0 0 - -")
  else
    {:ok, written} = Base64test.BytesTable.to_json_blob(value)
    IO.puts("0 #{report.clamped} #{report.kind_mismatch} #{hex.(value.payload)} #{hex.(written)}")
  end
end
